x/delayedack/keeper/invariants.go

```diff
@@ -0,0 +1,95 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	commontypes "github.com/dymensionxyz/dymension/v3/x/common/types"
+	"github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+)
+
+// RegisterInvariants registers the bank module invariants
+func RegisterInvariants(ir sdk.InvariantRegistry, k Keeper) {
+	ir.RegisterRoute(types.ModuleName, "rollapp-finalized-packet", RollappFinalizedPackets(k))
+	ir.RegisterRoute(types.ModuleName, "rollapp-reverted-packet", RollappRevertedPackets(k))
+}
+
+// AllInvariants runs all invariants of the X/bank module.
+func AllInvariants(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		res, stop := RollappFinalizedPackets(k)(ctx)
+		if stop {
+			return res, stop
+		}
+		res, stop = RollappRevertedPackets(k)(ctx)
+		if stop {
+			return res, stop
+		}
+		return "", false
+	}
+}
+
+// RollappFinalizedPackets checks that all rollapp packets stored for a rollapp finalized height are also finalized
+func RollappFinalizedPackets(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var msg string
+		packets := k.GetAllRollappPackets(ctx)
+
+		for _, packet := range packets {
```

Comment:
not sure why running the rollapp specific code per packet.
IMO it more clearer to get the finalized hight of each rollapp once, and then iterate over the packets and compare

Reply:
updated